kotlin.kt

```diff
@@ -1,20 +1,16 @@
-public fun main(args: Array<String>) {
-	for (i in 1..100) {
-		fizzBuzz(i)
-	}
+inline fun Int.divisibleBy(x: Int) = this % x == 0
+
+fun main(args: Array<String>) {
+	(1..100).forEach(::fizzBuzz)
 }
 
 fun fizzBuzz(n: Int) {
-	val divisibleBy3 = n % 3 == 0
-	val divisibleBy5 = n % 5 == 0
-
-	if (divisibleBy3 && divisibleBy5) {
-		println("FizzBuzz")
-	} else if (divisibleBy5) {
-		println("Buzz")
-	} else if (divisibleBy3) {
-		println("Fizz")
-	} else {
-		println(n)
-	}
+	println(
+		when (true) {
+			n.divisibleBy(15) -> "FizzBuzz"
```

Comment:
Oh my gosh, is it... if the expression on the left evaluates to x, for `when(x)` then the expression on the right is the output.

Then, is there a way to get around comparing them directly to a boolean value? It's a bit like saying `if (thisVar == true)` which is not pretty when you can just `if (thisVar)` if, however it's the most pragmatic way to do it. Perhaps leave a comment?